website/project/model.py

```diff
@@ -1519,11 +1551,7 @@ def register_node(self, schema, auth, template, data):
         :param auth: All the auth information including user, API key.
         :template: Template name
         :data: Form data
-
         """
-        if not self.can_edit(auth):
```

Comment:
Considering how important permissions are for this particular method, I revise my stance on this; better to have these auth-related business rules in the model as well, even at the cost of redundancy. Let's just keep the check here for now. We can revisit the other methods in the future.